### ISSUE TITLE
Whice piece of the OpenCV source should be modified to read Y10B in default?

### ISSUE THREAD

**Issue by lucyking** [TimboInSpace/Armin#2]
Hi Tim Sullivan,
As you say that the OpenCV source has been modified && reinstalled ,then may you show that which && how the particular source should be modified? :-)

ps:as for **"file:///home/pi/kinect_calibration.yaml"**, is it looks like [this](https://svn.personalrobotics.ri.cmu.edu/public/13.1/moped/moped-modeling-py/cam_examples/kinect_calibration.yaml)? 

**pps**:sorry, I've see the issue in the librekinect respority just now, I will try the img which you upload and see if everything works well. 

In the  #1 issue,  tiancovici's board is Raspberry Pi 2.There're some differences between  Raspberry Pi and Raspberry Pi 2 ,so may you let me know which board you used?

**Comment by TimboInSpace** [comment]
Yep. It was a long time ago but I'll try to find it and post a link.

**Comment by TimboInSpace** [comment]
For the calibration yaml, your task is to define the "camera matrix". See the opencv docs:
http://docs.opencv.org/modules/calib3d/doc/camera_calibration_and_3d_reconstruction.html

I highly recommend generating your own calibration file from camera_calibration.py, if you can. I did mine manually because the calibration script had trouble opening my 640x40 stream.

Given it's a 640x480 stream, your yaml file looks totally valid. Very similar to mine before I modified it for 640x40

**Comment by TimboInSpace** [comment]
I just downloaded a fresh copy of the opencv 2.3.1 source, I'll run _diff_ over this fresh folder and my custom source code, and see which files were modified. (I had no luck in finding the modification procedure from my browser history or file system timestamps)

**Comment by TimboInSpace** [comment]
It appears the only change is to OpenCV-2.3.1/modules/highgui/src/cap_libv4l.cpp
Changed this:
#define DEFAULT_V4L_HEIGHT 480
to this:
#define DEFAULT_V4L_HEIGHT 488

I thought there was more required, but this is all that diff is reporting

**Comment by lucyking** [comment]
Many thanks to your first hand information && detailed description.
I will try to modify the *.yaml file manually next.
If there is good news I will inform you in time ;-)

**Comment by lucyking** [comment]
Hi Tim,i've downloaded your img already.
But according your PDF's intruduction,we should contact with the Kinect through PC's shell command && Python script.This semms not portable&&convenient..

Then I've make a new project using 【Raspberry Pi 2+[OpenNI](https://github.com/OpenNI/OpenNI)+[Sensor](https://github.com/ruedigerH2/SensorKinect)】to get the depth data without monitor recently.The speed is about 12 frame/s(in 640*480) and works well.If my code may do a favor to you, just let me know :-)

And your NOTES' PDF i've got that we can "get depth_image in shrunken 640x40 reslution",so we just get a slim part of the whole depth_image per frame,or collect many parts to generate a complete one ?May you upload a example picture to show the depth_image that you got ?

**Comment by TimboInSpace** [comment]
Portable? Well I meant for this image to be a "write image to SD, plug it
in, and run it" solution, for the very specific problem of getting a Kinect
depthimage from a raspberry pi to ROS - so you're right it's not portable
at all, but I hadn't really intended it to be. I would say it's very
convenient though, if you're trying to solve the same problem as me.

As for the 40 pixel height, I did that do decrease bandwidth over wifi. It
was not intended to reassemble the image from the 40 pixel strips, just to
truncate the image and send the wide strip of data I needed to produce a
psuedo-laserscan. It helped quite a bit, but not as much as I had hoped:
I'm able to do the ROS stuff I need to, but only get 6fps as most from the
Kinect. Before I reduced the frame height, my laptop which receives the
depthimages complained about receiving incomplete frames.

That's excellent news about your implementation using OpenNi2! I have an
alternate raspi configuration that I test things on, and tried to use
OpenNi2 on that system: In the end I could not get OpenNi to detect that
the kinect was plugged in ( I was using a powered USB hub, and installed
libusb from source, still no luck). So YES absolutely, I would like your
code! My current setup is quite flawed, and the robot would perform a lot
better with the higher bit depth, frame size, and frame rate that OpenNi2
provides.

**Comment by TimboInSpace** [comment]
Just let me know how you want to send your code. Also, a walkthrough of how
you installed OpenNI2 to the raspberry pi would be very appreciated :D I'm
wondering if maybe it's the fact that you're using a Raspberry Pi 2, if our
installation procedures were the same.

**Comment by lucyking** [comment]
Hi Tim,
Yes,I using a  Pi 2  because **Pi 1's poor hardware can't deal with Kinect at all**.  
With **portable** what information I want to deliver is  that the Raspberry PI can not deal with the depth_image by itself but should cantact with PC through WiFi. It seems **inconvenient** and the robot can't response to the envirenment as soon as possible.I may try using more appropriate words later :-) 
I will push my [code](https://github.com/lucyking/Kinect_Depth) to github sooner and the sdcard img to a Cloud Drive somewhere.
The img has everything built allready, what we have to do is just plug it in then power up the Pi 2 with the connected Kinect.

**Comment by TimboInSpace** [comment]
Ah yes I see what you mean. In that sense it is probably the least portable software ever :P 
I'm using the roomba's bump and wall-detection sensors for immediate response to the environment, the kinect data is for visual odometry and localization - so I didn't need real-time performace there.

I just unboxed my raspberry pi 2! I'll give it a try using my previously-installed libfreenect.
My biggest question (because it affects the power distribution for my robot) is: did you need a powered usb hub, or are you connecting the kinect directly into the Pi's usb?

**Comment by TimboInSpace** [comment]
Having some trouble installing SensorKinect. The Redist  folder isn't being created when running RedistMaker.

**Comment by TimboInSpace** [comment]
Just got it working, I have the console output from OpenNI running. Now I'll try to access the stream with ROS somehow

**Comment by lucyking** [comment]
I just connect the Kinect directly to Pi.Because the kinect has its own AC Adapter.
When I use old boards such as BBB or Raspberry Pi B+,although with powered USB Hub,they do not work well all the same.
Glad to hear you've got it work.And with Pi 2's better hardware,may you also can running your previous code perfectly : - ）

**Comment by TimboInSpace** [comment]
Could you send your code for reading the kinect when you get a chance?  Right now I'm trying to integrate it into ROS and having no success. However if that is not a good option, there are ways to circumvent ROS, and would likely be very very similar to your code

**Comment by lucyking** [comment]
here's my code:https://github.com/lucyking/Kinect_Depth
what is that you integrate into ROS failed?

**Comment by TimboInSpace** [comment]
ROS has it's own OpenNI stack that has all of the code to communicate with other ROS nodes pre-written. I couldn't get that to work, but wrote my own to publish the ROS messages.
It's working great on the raspberry pi side (publishing two things: the image, and image metadata), but on my base computer it appears to be receiving incomplete messages. It then drops the incomplete ones, and complains that the images and metadata are out of sync.

I ended up adapting the SimpleRead example to open the depth frame as a OpenCV mat, then publish it as an ROS message.
